api/src/routing.js

```diff
@@ -161,6 +161,8 @@ app.use(
           `'sha256-B5cfIVb4/wnv2ixHP03bHeMXZDszDL610YG5wdDq/Tc='`,
           // AngularJS and several dependencies require this
           `'unsafe-eval'`,
+          // Angular styles optimization requires this because it adds an inline onload event handler
+          `'sha256-MhtPZXr7+LpJUY5qtMutB+qWfQtMaPccfe7QXtCcEYc='`,
```

Comment:
The comment here helps to understand what's going on, but in the case anyone needs to update this in the future, is it generated automatically via the ng tooling or does a future person need to know to do something? If the latter, please add a note about how to do that, ideally just a url to docs/instructions.

Reply:
This is not generated automatically. It turns out the Angular upgrade guide does not mention this and so far I haven't found it in any of their changelogs.  
I figured it out by looking at the line the browser was complaining about and later found out other people had the same problem: https://github.com/angular/angular-cli/issues/20864

I could add that issue on the comment as it describes why it broke and comments contain alternatives to our solution.

Reply:
I got focused on the specific change here but lost sight of what the feature actually is. Cases for "critical css" are usually when a user loads a page in their browser and gets the [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content), largely because a separate http request has not yet completed sooner in the page load lifecycle. That doesn't sound like it applies to our use case where all resources are available locally already. Do we need to use "critical css" at all?

Please excuse my ignorance if this is a previously-resolved discussion and I just need to catch up!

Reply:
You're completely right, offline users already have these static files locally and they will only benefit from this change on the very first load or when they get an update.  
This mainly benefits online users who don't have the service worker to cache those resources locally.

Reply:
Online users also use the service-worker. It gets installed on login for all users: https://github.com/medic/cht-core/blob/master/api/src/public/login/script.js#L212

The part that online users don't have is the bootstrap part, where the app only loads _after_ a successful service worker registration.

Reply:
Thanks for clearing that up Diana! :+1: